### PR TITLE
Support the 'current' global object.

### PR DESCRIPTION
**Pull request by Ms2ger**
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/14647)

### CODE REVIEW

**Comment by jdm** [comment]
On the surface this seems like a partial implementation of the script settings stack, but I don't see how we could build a stack that we could return non-current entries from. Since that's the case, it's not clear to me why it's worth implementing this by itself, rather than doing the script stack properly.

**Comment by Ms2ger** [comment]
I don't see a reason to block this on a full implementation of incumbent/entry globals; there will be no need to change this code when we do that, and at least now we can properly implement the security checks for `contentDocument`.

**Comment by metajack** [comment]
cc @Ms2ger

**Comment by jdm** [comment]
That makes sense.

**Comment by metajack** [comment]
cc @metajack

**Comment by Ms2ger** [comment]
Depends on https://github.com/servo/rust-mozjs/pull/322

File: components/script/dom/xmlhttprequest.rs [hunk 1]
```diff
@@ -1204,6 +1204,7 @@ impl XMLHttpRequest {
         Document::new(win,
                       None,
                       parsed_url,
+                      doc.origin().alias(),
```
Comment (jdm) [inline components/script/dom/xmlhttprequest.rs]: Given that the XHR specification doesn't specify any origin, I believe this should be an opaque origin instead.

**Review by jdm — CHANGES_REQUESTED** [review]

File: components/script/dom/xmlhttprequest.rs (hunk 1 above)
Comment (Ms2ger) [inline components/script/dom/xmlhttprequest.rs]: > Set document's origin to context object's relevant settings object's origin. 

https://xhr.spec.whatwg.org/#document-response step 11

**Review by jdm — APPROVED** [review]

**Comment by Ms2ger** [comment]
```
Tests with unexpected results:
  ▶ ERROR [expected OK] /workers/baseurl/alpha/sharedworker.html
  └   → SharedWorker is not defined
```

I'm not sure how that could happen; I'll try to reproduce locally first.

**Comment by Ms2ger** [comment]
@bors-servo r- try retry

Doesn't reproduce locally. Let's see if it's consistent on the builder.

**Comment by Ms2ger** [comment]
I ran this 800 times locally and didn't fail once; filed https://github.com/servo/servo/issues/14772.

@bors-servo try- r=jdm retry

**Pull request merged**